### PR TITLE
Issue 2109: TextInput - remove keyboard handlers when unmounting.

### PR DESCRIPTION
**Pull request by RobertGary1**

### CODE REVIEW

File: src/js/components/TextInput.js [hunk 1]
```diff
@@ -41,18 +41,7 @@ export default class TextInput extends Component {
...
-      tab: this._onRemoveDrop,
-      up: this._onPreviousSuggestion,
-      down: this._onNextSuggestion,
-      enter: this._onEnter
-    };
-    const focusedKeyboardHandlers = {
-      down: this._onAddDrop
-    };
-
+    const {activeKeyboardHandlers, focusedKeyboardHandlers} = this;
```
Comment (alansouzati) [inline src/js/components/TextInput.js]: Great contribution. I think we can simplify it a little bit:

```
export default class TextInput extends Component {
  activeKeyboardHandlers =  {
      esc: this._onRemoveDrop,
      tab: this._onRemoveDrop,
      up: this._onPreviousSuggestion,
      down: this._onNextSuggestion,
      enter: this._onEnter
  };

  focusedKeyboardHandlers =  {
      down: this._onAddDrop
  };

  componentDidUpdate (prevProps, prevState) {
      // use `this.activeKeyboardHandlers` and `this.focusedKeyboardHandlers`
  } 

  componenWillUnmount() {
     // use `this.activeKeyboardHandlers` and `this.focusedKeyboardHandlers`
  }
}
```

Comment (RobertGary1) [reply src/js/components/TextInput.js]: I think syntactically you have to include the get directive on them in ES6. I'm not sure you can directly declare class variables (although I've seen it suggested in ES7). Do we have examples of this being done in Grommet today?
eslint gives an error unless I put it below the lifecycle methods. You can disable this in eslint but I don't see it disabled in the eslint file.

**Comment by alansouzati** [comment]
it's been a while since I don't touch v1, but after a careful analysis you can actually follow the example of [DateTimeDrop](https://github.com/grommet/grommet/blob/master/src/js/components/DateTimeDrop.js#L73). Then we have a consistent solution for v1.

File: src/js/components/TextInput.js (hunk 1 above)
Comment (RobertGary1) [reply src/js/components/TextInput.js]: Ok, it should be up to date now

**Comment by alansouzati** [comment]
Thanks for your contribution.

**Pull request merged**